### PR TITLE
Add eslint linting

### PR DESCRIPTION
**Pull request by avahowell**
This PR implements #227.  Right now, everything is ignored except the new wallet plugin.  As remaining code is refactored it will be whitelisted in .eslintignore.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,11 +6,14 @@
...
     "chai": "^3.4.1",
     "chai-as-promised": "^5.1.0",
     "electron-mocha": "^0.7.0",
     "electron-packager": "^7.0.2",
     "electron-prebuilt-compile": "^1.2.0",
+    "eslint": "^2.11.0",
+    "eslint-plugin-react": "^5.1.1",
     "ink-docstrap": "^1.0.3",
     "jsdoc": "^3.4.0",
     "jshint": "^2.9.1-rc1"
```
Comment (lukechampine) [inline package.json]: can we get ride of some of these dependencies now? Specifically `chai`, `chai-as-promised`, `electron-mocha`, `ink-docstrap`, `jsdoc`, and `jshint`?

Comment (avahowell) [inline package.json]: yes.  I'm going to leave `chai` since I plan to use it to test the Wallet plugin (and the plugin API/new main UI stuff), but the rest can go.

File: plugins/Wallet/js/components/receiveprompt.js [hunk 2]
```diff
@@ -1,15 +1,18 @@
 import React, { PropTypes } from 'react'
 
-const ReceivePrompt = ({address, actions}) => (
-	<div className="modal">
-		<div className="receive-prompt">
-			You can receive Siacoins using the following address:
-			<div className="wallet-address">{ address }</div>
-			<button className="receiveprompt-dismissbtn" onClick={actions.hideReceivePrompt}>OK</button>
+const ReceivePrompt = ({address, actions}) => {
+	const handleDismissClick = () => actions.hideReceivePrompt()
```
Comment (lukechampine) [inline plugins/Wallet/js/components/receiveprompt.js:4]: why was this changed to a standalone function? Are we standardizing on "`onClick` handlers should be defined in the Component body?"

Comment (avahowell) [inline plugins/Wallet/js/components/receiveprompt.js:4]: Yes.  Right now we're not doing anything except dispatching the action, but in the future we could do more (validation for example).  This is also enforced by `"react/jsx-handler-names": 1` in eslintrc.  The alternative would be to rename all click handler actions with the `handle` prefix, or decide on a different prefix to be standard.
